src/main/java/io/confluent/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -232,6 +234,9 @@ private long executeDeletes() throws SQLException {
     long totalDeleteCount = 0;
     if (nonNull(deletePreparedStatement)) {
       for (int updateCount : deletePreparedStatement.executeBatch()) {
+        if (updateCount == Statement.EXECUTE_FAILED) {
+          throw new BatchUpdateException("Execution failed for part of the batch delete", new int[]{Statement.EXECUTE_FAILED});
```

Comment:
same comment as above. Requesting to pass through `deletePreparedStatement.executeBatch()` instead of creating a new int array with `Statement.EXECUTE_FAILED`